Feature/task 1

Hakonkk opened the pull request.

Hakonkk committed:
TASK-1 Поменял проперти в yaml

Hakonkk committed:
Добавил слои (интеграционный,сервис,репозиторий,конвертор контроллер,маппер)
проверил получение заданного количества рандомных пользователей, добавил зависемость mapstruct.

github1901zx commented on src/main/java/com/example/sergpetproject/SergPetProjectApplication.java:
```diff
@@ -1,13 +1,16 @@
 
     public static void main(String[] args) {
-        SpringApplication.run(SergPetProjectApplication.class, args);
+//        SpringApplication.run(SergPetProjectApplication.class, args);
```
Убрать комментарии если они не несут смысла

Hakonkk replied:
тестировал метод, исправил

github1901zx commented on src/main/java/com/example/sergpetproject/controllers/RandomApiController.java:
```diff
@@ -0,0 +1,20 @@
+    final RandomApiResponse randomApiResponse;
+    final RandomApiHttpClient randomApiHttpClient;
+
+    public RandomApiController(RandomApiResponse randomApiResponse, RandomApiHttpClient randomApiHttpClient) {
```
@RequiredArgsConstructor](https://www.baeldung.com/java-lombok-constructor-annotation)

Hakonkk replied:
использовал @Autowired для инициализации в будущем, изменил на конструктор, вернуть как буду делать контроллер

github1901zx commented on src/main/java/com/example/sergpetproject/integration/RandomApiHttpClient.java:
```diff
@@ -0,0 +1,31 @@
+
+public class RandomApiHttpClient {
+    private static final String RANDOM_API_URL = "https://randomuser.me/api/?results=%s";
+    private final ObjectMapper mapper = new ObjectMapper();
```
@autowired или ижектировать через конструктор , поднять бин в @Config

Hakonkk replied:
тут не понял

github1901zx commented on src/main/java/com/example/sergpetproject/integration/RandomApiHttpClient.java:
```diff
@@ -0,0 +1,31 @@
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class RandomApiHttpClient {
```
Возможно перейти на FiengClient?

Hakonkk replied:
без понятия о чем речь

github1901zx commented on src/main/java/com/example/sergpetproject/repository/PersonRepository.java:
```diff
@@ -0,0 +1,7 @@
+
+import com.example.sergpetproject.integration.entity.PersonEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
@repository

Hakonkk replied:
исправил

github1901zx commented on src/main/java/com/example/sergpetproject/service/RandomApiServiceImpl.java:
```diff
@@ -0,0 +1,21 @@
+
+    @Override
+    public void saveAllPersons() {
+
```
а че пустой?

Hakonkk replied:
в процессе разработки

github1901zx replied:
поставь todo

Hakonkk committed:
Настроил Докер, создал и подключился к базе posgres через DBeaver
- добавил SQL-скрипт для создания таблиц persons, names, locations и streets и создал таблицы
- добавил тест для проверки пола в классе PersonTest
- изменил application.yaml для подключения к базе данных PostgreSQL
- отформатировал код в классе JsonNodeToRandomApiResponse
- удалил закомментированный код в классе SergPetProjectApplication
```

Hakonkk committed:
- Добавить маппинг из RandomApiResponse в PersonEntity с помощью MapStruct
- Добавить аннотации Lombok для сущностей Name, Location и Street
- Добавил метод saveAllPersons в RandomApiServiceImpl, который сохраняет случайных людей в базу данных
- Убрал лишние поля и конструкторы из RandomApiController
- Добавил аннотацию @Service и конструкторы для RandomApiHttpClient
- Вызвал метод saveAllPersons в SergPetProjectApplication для проверки работы сервиса

Hakonkk committed:
добавил методы в сервисный слой save get delete, в контроллере вызвал методы, get метод работает, save метод не работает, ошибка с БД пишет, что таблица streets не найдена, хотя она есть, проверил связи, правописание, наличие, удалил, пересоздал, все-равно пишет что ее нет

Hakonkk committed:
Добавил ddl-auto в yaml для правильной работы скриптов БД
написал метод deleteById для удаления person из БД по id проверил работоспособность всех 3-х методов в postman
